deepcell/image_generators/semantic.py

```diff
@@ -385,3 +375,360 @@ def random_transform(self, x, y=None, seed=None):
+    """
+
+    def __init__(self,
+                 train_dict,
+                 movie_data_generator,
+                 batch_size=1,
+                 frames_per_batch=5,
+                 shuffle=False,
+                 transforms=['watershed-cont'],
+                 transforms_kwargs={},
+                 seed=None,
+                 min_objects=3,
+                 data_format='channels_last'):
+        X, y = train_dict['X'], train_dict['y']
+        if X.shape[0] != y.shape[0]:
+            raise ValueError('Training batches and labels should have the same'
+                             'length. Found X.shape: {} y.shape: {}'.format(
+                                 X.shape, y.shape))
+
+        if X.ndim != 5:
+            raise ValueError('Input data in `SemanticMovieIterator` '
+                             'should have rank 5. You passed an array '
+                             'with shape', self.x.shape)
+
+        if y is None:
+            raise ValueError('Instance masks are required for the '
+                             'SemanticIterator')
+
+        self.x = np.asarray(X, dtype=K.floatx())
+        self.y = np.asarray(y, dtype='int32')
+        self.frames_per_batch = frames_per_batch
+        self.transforms = transforms
+        self.transforms_kwargs = transforms_kwargs
+        self.channel_axis = 4 if data_format == 'channels_last' else 1
+        self.time_axis = 1 if data_format == 'channels_last' else 2
+        self.row_axis = 2 if data_format == 'channels_last' else 3
+        self.col_axis = 3 if data_format == 'channels_last' else 4
+        self.movie_data_generator = movie_data_generator
+        self.data_format = data_format
+        self.min_objects = min_objects
+
+        self.y_semantic_list = []  # optional semantic segmentation targets
+
+        # Create a list of all the semantic targets. We need to be able
+        # to have multiple semantic heads
+        # Add all the keys that contain y_semantic
+
+        # Add transformed masks
+        for transform in transforms:
+            transform_kwargs = transforms_kwargs.get(transform, dict())
+            y_transform = _transform_masks(y, transform,
+                                           data_format=data_format,
+                                           **transform_kwargs)
+            if y_transform.shape[self.channel_axis] > 1:
+                y_transform = np.asarray(y_transform, dtype='int32')
+            elif y_transform.shape[self.channel_axis] == 1:
+                y_transform = np.asarray(y_transform, dtype=K.floatx())
+            self.y_semantic_list.append(y_transform)
+
+        invalid_batches = []
+
+        # Remove images with small numbers of cells
+        for b in range(self.x.shape[0]):
+            y_batch = np.squeeze(self.y[b], axis=self.channel_axis - 1)
+            y_batch = np.expand_dims(y_batch, axis=self.channel_axis - 1)
+
+            self.y[b] = y_batch
```

Comment:
Why are we doing this? It looks like we just squeeze out the channel and add it back in again? Does it affect the next lines, `< min_objects`?